hedera-node/src/test/java/com/hedera/services/state/merkle/MerkleAccountStateTest.java

```diff
@@ -411,6 +411,48 @@ void deserializeV0220Works() throws IOException {
 		assertEquals(subject, newSubject);
 	}
 
+	@Test
+	void deserializeV0230Works() throws IOException {
+		final var in = mock(SerializableDataInputStream.class);
+		subject.setNftsOwned(nftsOwned);
+		subject.setTokenAssociationMetadata(EMPTY_TOKEN_ASSOCIATION_META);
+		final var newSubject = new MerkleAccountState();
+		given(serdes.readNullable(argThat(in::equals), any(IoReadingFunction.class))).willReturn(key);
+		given(in.readLong())
```

Comment:
This test can be prone to errors . If the provided format here is not be correct to begin with (or if modified in a subsequent change to become invalid), we might end up writing implementation code for the wrong format. I'd suggest doing the following instead:

1. Check out the v0.23 source code. Write a quick tool to generate a serialized binary string for a known set of inputs.
2. Hardcode this as a byte array here, or in a file that can be loaded.
3. Hardcode the expected values.
4. Generate the same byte array for v0.25 and add to this test. Subsequent changes with new version numbers would then make use of this.

Reply:
[Done](https://github.com/hashgraph/hedera-services/blob/fix-0250-serde-bugs/hedera-node/src/test/java/com/hedera/services/state/merkle/MerkleAccountStateTest.java#L173)! Can you think of a good way to either automate such steps, or socialize them effectively? 😁

Reply:
This looks good for now. I can work on a refactor in a subsequent followup:

https://github.com/hashgraph/hedera-services/issues/3114